src/main/java/core/stack/MateStack.java

```diff
@@ -1,19 +1,55 @@
 package core.stack;
 
+import java.util.EmptyStackException;
+
 public class MateStack<T> {
-    public void push(T value) {
+    private static final int DEFAULT_SIZE = 8;
+    private Object[] stack;
+    //private T[] stack;
+    //Type parameter 'T' cannot be instantiated directly
+    private int size = 0;
+    private final int multiplier = 2;
+
+    public MateStack() {
+        stack = new Object[DEFAULT_SIZE];
+    }
 
+    public void grow() {
+        if (size < stack.length) {
+            int newCapacity = size * multiplier;
+            System.arraycopy(stack, 0, stack = new Object[newCapacity], 0, size);
```

Comment:
```suggestion
            System.arraycopy(stack, 0, stack, 0, size);
```

Reply:
why like this? if i do copy of exacly same array without growing its size then method grow is copletly usless. Also what about previeus comment? shall i change Object[] to T[] or not?